lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -540,6 +540,9 @@ export class UserAgentApplication {
 
       // prompt user for interaction
       this.navigateWindow(urlNavigate, popUpWindow);
+
+      const hash = await this.pollWindowForHash(popUpWindow);
+      this.handleAuthenticationResponse(hash);
```

Comment:
So shouldn't we restrict the handleAuthenticationResponse() call in the constructor now to only "redirect" cases?

Reply:
Yeah, good catch.

Reply:
Updated the constructor to check that we are not in an iframe, see above.